ФЗ 16

AlksSrg opened the pull request.

AlksSrg committed:
ФЗ 16

AlksSrg committed:
правки по чек стайл

AlksSrg committed:
добавил файлы докера

AlksSrg committed:
чек стайл

AlksSrg committed:
чек стайл

AlksSrg committed:
чек стайл

AlksSrg committed:
чек стайл

AlksSrg committed:
чек стайл

AlksSrg committed:
чек стайл

AlksSrg committed:
чек стайл

AlksSrg committed:
чек стайл

AlksSrg committed:
чек стайл

AlksSrg committed:
чек стайл

AlksSrg committed:
чек стайл

AlksSrg committed:
чек стайл

AlksSrg committed:
чек стайл

AlksSrg committed:
чек стайл

AlksSrg committed:
чек стайл

AlksSrg committed:
чек стайл

AlksSrg committed:
чек стайл

kriwsha commented on gateway/src/main/java/ru/practicum/shareit/item/controller/ItemController.java:
```diff
@@ -0,0 +1,128 @@
+                                              @RequestParam(defaultValue = "10") @Positive Integer size) {
+        // Проверяем, что текст не пустой
+        if (text == null || text.trim().isEmpty()) {
+            return ResponseEntity.ok().body("[]");
```
Почему возвращается ok, если текст пустой? Если это поведение правильное, то хардкодить строку пустого массива неверно: лучше тогда сам пустой массив или список положить

kriwsha commented on gateway/src/main/java/ru/practicum/shareit/item/controller/ItemController.java:
```diff
@@ -0,0 +1,128 @@
+     * @return список найденных вещей
+     */
+    @GetMapping("/search")
+    public ResponseEntity<Object> searchItems(@RequestParam String text,
```
Вообще не следует инициировать Object'ом. У вас ошибки обрабатываются отдельным контроллером, а здесь точно возвращается конкретный ответ. Так бы я рекомендовал поправить везде. Кстати, с учетом того, что у вас RestController, а не Controller, то вы можете обойтись без ResponseEntity

kriwsha commented on gateway/src/main/java/ru/practicum/shareit/request/dto/ItemRequestDto.java:
```diff
@@ -0,0 +1,43 @@
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class ItemRequestDto {
```
Где-то рекорд, где-то класс. Лучше выбрать единый стиль

AlksSrg committed:
Правки по замечаниям
1)Убрал "хардкор"
2)Поменял class на record + доработал некоторые тесты под record'ы
3)Пересмотрел использование аннотаций у client'ов и controller'ов

AlksSrg committed:
правки в Dockerfile

AlksSrg committed:
правки в Dockerfile

AlksSrg committed:
правки в Dockerfile

AlksSrg committed:
правки в Dockerfile

AlksSrg committed:
правки в docker-compose.yml

kriwsha commented on gateway/src/main/java/ru/practicum/shareit/booking/controller/BookingController.java:
```diff
@@ -0,0 +1,106 @@
+ * Выполняет валидацию входных данных и проксирует запросы на основной сервер.
+ */
+@Validated
+@Controller
```
На самом деле я имел ввиду как раз другое. Аннотация Controller устаревшая, на ее место пришла аннотация RestController. И использовать нужно RestController. При этом используя RestController не обязательно создавать ResponseEntity. Spring сам обернет ответ в энтити с кодом 200, если не указано иное. Поэтому давайте вернем аннотацию RestController

AlksSrg committed:
правки по замечаниям
